src/telliot_feeds/reporters/reporter360.py

```diff
@@ -0,0 +1,181 @@
+            "getStakerInfo",
+            _stakerAddress=self.acct_addr,
+        )
+
+        if (not status.ok) or (staker_info is None):
+            msg = "Unable to read reporters staker info"
+            return False, error_status(msg, log=logger.info)
+
+        if self.stake is None:
+            self.stake = stake
+        else:
+            msg = "Stake amount has changed due to TRB price change."
+            logger.info(msg)
+
+        if self.staker_info is None:
+            self.staker_info = StakerInfo(*staker_info)
+        elif self.staker_info.stake_balance > staker_info[1]:
+            # update balance after depositing more stake
+            self.staker_info.stake_balance = staker_info[1]
+            logger.info("your staked balance has decreased and account might be in dispute")
+
+        account_staked_bal = self.staker_info.stake_balance
+
+        if self.stake > account_staked_bal:
+            msg = (
+                "Current stake is low. "
+                # Add more TRB to continue reporting
+                + "Approving & depositing stake."
+            )
+            logger.info(msg)
+
+            gas_price_gwei = await self.fetch_gas_price()
+            if gas_price_gwei is None:
+                return False, error_status("Unable to fetch matic gas price for staking", log=logger.info)
+
+            # diff in required stake and account staked balance
+            stake_diff = self.stake - account_staked_bal
+
+            # check TRB wallet balance!
+            wallet_balance = self.token.read("balanceOf", account=self.acct_addr)
+            logger.info(f"Current wallet TRB balance: {wallet_balance}")
+
+            if stake_diff > wallet_balance:
+                msg = "Not enough TRB in the account to cover the stake"
+                return False, error_status(msg, log=logger.warning)
+
+            txn_kwargs = {"gas_limit": 300000, "legacy_gas_price": gas_price_gwei, "_amount": stake_diff}
+            # approve token spending
+            _, approve_status = await self.token.write(func_name="approve", spender=self.oracle.address, **txn_kwargs)
+            if not approve_status.ok:
+                msg = "Unable to approve staking"
+                return False, error_status(msg, log=logger.error)
+
+            # deposit stake
+            _, deposit_status = self.oracle.write("depositStake", **txn_kwargs)
+            if not deposit_status.ok:
+                msg = (
+                    "Unable to stake deposit: "
+                    + deposit_status.error
+                    + f"Make sure {self.acct_addr} has enough of the current chain's "
+                    + "currency and the oracle's currency (TRB)"
+                )
+                return False, error_status(msg, log=logger.error)
+
+        return True, ResponseStatus()
+
+    async def check_reporter_lock(self) -> ResponseStatus:
+
+        account_staked_bal = self.staker_info.stake_balance  # type: ignore
```

Comment:
I'm assuming the multiple typing ignores are because those attributes might be `None`. I think we should check for that and return and error_status if that's the case.